src/DFApplication.cpp

```diff
@@ -39,15 +44,144 @@ __reg__("DFApplication", [] (const SmartDaqApplication* smartApp,
 
-  // Code to actually generate modules should go here
-  std::cout << __FUNCTION__ << " -- Unimplemented function\n";
+  // Containers for module specific config objects for output/input
+  // Prepare TRB output objects 
+  std::vector<const oksdbinterfaces::ConfigObject*> trbOutputObjs;
+
+  // -- First, we process expected Queue and Network connections and create their objects.
+
+  // Process the queue rules looking for the TriggerRecord queue between TRB and DataWriter
+  const QueueDescriptor* trQDesc = nullptr;
+  for (auto rule : get_queue_rules()) {
+    auto destination_class = rule->get_destination_class();
+    if (destination_class == "DataWriter") {
+      trQDesc = rule->get_descriptor();
+    }
+  }
+  if (trQDesc == nullptr) { // BadConf if no descriptor between TRB and DataWriter
+    throw (BadConf(ERS_HERE, "Could not find queue descriptor rule for TriggerRecords!"));
+  }
+  // Create queue connection config object
+  oksdbinterfaces::ConfigObject trQueueObj;
+  std::string trQueueUid("trigger-records-"+UID());
+  confdb->create(dbfile, "Queue", trQueueUid, trQueueObj);
+  fill_queue_object_from_desc(trQDesc, trQueueObj);
+  // Place trigger record queue object into vector of output objs of TRB module
+  trbOutputObjs.push_back(&trQueueObj); 
+
+  // Process the network rules looking for the Fragments and TriggerDecision inputs for TRB
+  const NetworkConnectionDescriptor* fragNetDesc = nullptr;
+  const NetworkConnectionDescriptor* trigdecNetDesc = nullptr; 
+  for (auto rule : get_network_rules()) {
+    auto endpoint_class = rule->get_endpoint_class();
+    auto descriptor = rule->get_descriptor();
+    auto data_type = descriptor->get_data_type();
+    if (endpoint_class == "TRBuilder" && data_type == "Fragment") {
+      fragNetDesc = rule->get_descriptor();
+    }
+    else if (endpoint_class == "TRBuilder" && data_type == "TriggerDecision") {
+      trigdecNetDesc = rule->get_descriptor();
+    }
+  }
+  if (fragNetDesc == nullptr) { // BadConf if no descriptor for Fragments into TRB
+    throw (BadConf(ERS_HERE, "Could not find network descriptor rule for input Fragments!"));
+  }
+  if (trigdecNetDesc == nullptr) { // BadCond if no descriptor for TriggerDecisions into TRB
+    throw (BadConf(ERS_HERE, "Could not find network descriptor rule for input TriggerDecisions!"));
+  }
+  // Create network connection config object
+  oksdbinterfaces::ConfigObject fragNetObj;
+  oksdbinterfaces::ConfigObject trigdecNetObj;
+  std::string fragNetUid("fragments-to-dataflow-"+UID());
+  std::string trigdecNetUid("tigger-decisions-"+UID());
+  confdb->create(dbfile, "NetworkConnection", fragNetUid, fragNetObj);
+  confdb->create(dbfile, "NetworkConnection", trigdecNetUid, trigdecNetObj);
+  fill_netconn_object_from_desc(fragNetDesc, fragNetObj);
+  fill_netconn_object_from_desc(trigdecNetDesc, trigdecNetObj);
+
+  // Process special Network rules! 
+  // Looking for DataRequest rules from ReadoutAppplications in current Session
```

Comment:
I don't know why the TRB needs to parse the Readout apps' configuration to find where to send requests. Shouldn't it just be able to get a list of all "DataRequest" endpoints from the connectivity service. If you really have to parse the Readout app configurations, you should check the enabled/disabled state of the app before adding its endpoint.